packages/components/src/components/domainproperties/dataclasses/DataClassPropertiesPanel.tsx

```diff
@@ -0,0 +1,208 @@
+            <DomainPropertiesPanelProvider
+                controlledCollapse={controlledCollapse}
+                collapsible={collapsible}
+                initCollapsed={initCollapsed}
+                onToggle={onToggle}
+            >
+                <DataClassPropertiesPanelImpl {...this.props} />
+            </DomainPropertiesPanelProvider>
+        )
+    }
+}
+
+class DataClassPropertiesPanelImpl extends React.Component<Props, State> {
+
+    static defaultProps = {
+        noun: 'Data Class',
+        appPropertiesOnly: false,
+        initCollapsed: false,
+        validate: false,
+        helpTopic: DEFINE_DATA_CLASS_TOPIC
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isValid: true
+        };
+    }
+
+    componentWillReceiveProps(nextProps: Readonly<Props>, nextContext: any): void {
+        const { validate } = this.props;
+        if (nextProps.validate && validate !== nextProps.validate) {
+            this.setIsValid();
+        }
+    }
+
+    componentDidMount(): void {
+        updateDomainPanelClassList(this.props.useTheme, undefined, 'dataclass-properties-hdr');
+    }
+
+    componentDidUpdate(prevProps: Readonly<Props>, prevState: Readonly<State>, snapshot?: any): void {
+        updateDomainPanelClassList(prevProps.useTheme, undefined, 'dataclass-properties-hdr');
+    }
+
+    setIsValid() {
+        const { model, onChange } = this.props;
+        const isValid = model && model.hasValidProperties();
+        this.setState(() => ({isValid}), () => onChange(model));
+    }
+
+    toggleLocalPanel = (evt: any, context: IDomainPropertiesPanelContext): void => {
+        this.setIsValid();
+        context.togglePanel(evt, !context.collapsed);
+    };
+
+    onFormChange = (evt: any) => {
+        const id = evt.target.id;
+        const value = evt.target.value;
+        this.onChange(id, value);
+    };
+
+    onChange = (id: string, value: any) => {
+        const { model, onChange } = this.props;
+        onChange(model.set(getFormNameFromId(id), value) as DataClassModel);
+    };
+
+    renderSampleSetSelect() {
+        const { model, noun } = this.props;
+
+        return (
+            <Row>
+                <Col xs={3}>
+                    <LabelOverlay
+                        label={'Sample Set'}
+                        description={`The default Sample Set where new samples will be created for this ${noun.toLowerCase()}.`}
+                        canMouseOverTooltip={true}
+                    />
+                </Col>
+                <Col xs={9}>
+                    <QuerySelect
+                        componentId={FORM_IDS.MATERIAL_SOURCE_ID}
+                        name={FORM_IDS.MATERIAL_SOURCE_ID}
+                        schemaQuery={SCHEMAS.EXP_TABLES.SAMPLE_SETS}
+                        formsy={false}
+                        showLabel={false}
+                        preLoad={true}
+                        // fireQSChangeOnInit={true}
```

Comment:
Assuming these are commented out until relevant dev is completed? For clarity, should there be another comment that mentions, 'Uncomment when X'? (Just saw Binal use a similar pattern before)

Reply:
Yes, I will finish out this implementation in the next feature branch for the DataClassDesigner. I'll add another comment to that effect. Thanks.

Reply:
Done